test/unit/vagrant/util/guest_networks_spec.rb

```diff
@@ -53,95 +53,116 @@
+      expect(comm.received_commands).to include(%r{mv '/tmp/vagrant.*eth3.*' '/etc/NetworkManager/system-connections/eth3.nmconnection'})
+    end
 
-    expect(comm.received_commands).to include("nmcli d connect 'eth1'")
-    expect(comm.received_commands).to include("nmcli d connect 'eth2'")
-    expect(comm.received_commands).to include("nmcli d connect 'eth3'")
-  end
+    it "should move load new configuration files" do
+      subject.configure_network_manager(machine, [network_1, network_2, network_3])
 
-  context "network configuration file" do
-    let(:networks){ [[:public_network, network_1], [:private_network, network_2], [:private_network, network_3]] }
+      expect(comm.received_commands).to include("nmcli c load '/etc/NetworkManager/system-connections/eth1.nmconnection'")
+      expect(comm.received_commands).to include("nmcli c load '/etc/NetworkManager/system-connections/eth2.nmconnection'")
+      expect(comm.received_commands).to include("nmcli c load '/etc/NetworkManager/system-connections/eth3.nmconnection'")
+    end
 
-    let(:tempfile) { double("tempfile") }
+    it "should connect new devices" do
+      subject.configure_network_manager(machine, [network_1, network_2, network_3])
 
-    before do
-      allow(tempfile).to receive(:binmode)
-      allow(tempfile).to receive(:write)
-      allow(tempfile).to receive(:fsync)
-      allow(tempfile).to receive(:close)
-      allow(tempfile).to receive(:path)
-      allow(Tempfile).to receive(:open).and_yield(tempfile)
+      expect(comm.received_commands).to include("nmcli d connect 'eth1'")
+      expect(comm.received_commands).to include("nmcli d connect 'eth2'")
+      expect(comm.received_commands).to include("nmcli d connect 'eth3'")
     end
 
-    it "should generate two configuration files" do
-      expect(Tempfile).to receive(:open).twice
-      subject.configure_network_manager(machine, [network_1, network_2])
+    context "network configuration file" do
+      let(:networks){ [[:public_network, network_1], [:private_network, network_2], [:private_network, network_3]] }
+
+      let(:tempfile) { double("tempfile") }
+
+      before do
+        allow(tempfile).to receive(:binmode)
+        allow(tempfile).to receive(:write)
+        allow(tempfile).to receive(:fsync)
+        allow(tempfile).to receive(:close)
+        allow(tempfile).to receive(:path)
+        allow(Tempfile).to receive(:open).and_yield(tempfile)
+      end
+
+      it "should generate two configuration files" do
+        expect(Tempfile).to receive(:open).twice
+        subject.configure_network_manager(machine, [network_1, network_2])
+      end
+
+      it "should generate three configuration files" do
+        expect(Tempfile).to receive(:open).thrice
+        subject.configure_network_manager(machine, [network_1, network_2, network_3])
+      end
+
+      it "should generate configuration with network_2 IP address" do
+        expect(tempfile).to receive(:write).with(/#{network_2[:ip]}/)
```

Comment:
Escape the variable value used within the regular expression:

```ruby
.with(/#{Regexp.escape(network_2[:ip])}/)
```

I know these were existing, and just moved, but would be good to make them more correct while here :slightly_smiling_face: